src/Stratis.Bitcoin.Features.LightWallet/LightWalletSyncManager.cs

```diff
@@ -245,33 +256,57 @@ public void SyncFromDate(DateTime date)
+            {
+                this.logger.LogTrace("(-)[SYNCFROMDATEASYNCLOOP_NOT_NULL]");
+                return;
+            }
+
             // Before we start syncing we need to make sure that the chain is at a certain level.
             // If the chain is behind the date from which we want to sync, we wait for it to catch up, and then we start syncing.
             // If the chain is already past the date we want to sync from, we don't wait, even though the chain might not be fully downloaded.
-            if (this.chain.Tip.Header.BlockTime.LocalDateTime < date)
+            if (this.chain.Tip.Header.BlockTime.LocalDateTime < this.syncWalletDate)
             {
-                this.logger.LogTrace("The chain tip's date ({0}) is behind the date from which we want to sync ({1}). Waiting for the chain to catch up.", this.chain.Tip.Header.BlockTime.LocalDateTime, date);
+                this.logger.LogTrace("The chain tip's date ({0}) is behind the date from which we want to sync ({1}). Waiting for the chain to catch up.", this.chain.Tip.Header.BlockTime.LocalDateTime, this.syncWalletDate);
 
-                this.asyncLoop = this.asyncLoopFactory.RunUntil("LightWalletSyncManager.SyncFromDate", this.nodeLifetime.ApplicationStopping,
-                    () => this.chain.Tip.Header.BlockTime.LocalDateTime >= date,
+                this.syncFromDateAsyncLoop = this.asyncLoopFactory.RunUntil("LightWalletSyncManager.SyncFromDate", this.nodeLifetime.ApplicationStopping,
+                    () => this.chain.Tip.Header.BlockTime.LocalDateTime >= this.syncWalletDate,
                     () =>
                     {
-                        this.logger.LogTrace("Start syncing from {0}.", date);
-                        this.StartSync(this.chain.GetHeightAtTime(date));
+                        int blockHeightAtDate = this.chain.GetHeightAtTime(this.syncWalletDate);
+                        this.logger.LogTrace("Start syncing from {0} (block: {1}).", this.syncWalletDate, blockHeightAtDate);
+                        SyncFromHeight(blockHeightAtDate);
+
+                        // Set the asyncloop to null so that it can be used again.
+                        this.syncFromDateAsyncLoop = null;
                     },
                     (ex) =>
                     {
-                        // in case of an exception while waiting for the chain to be at a certain height, we just cut our losses and
-                        // sync from the current height.
+                        // In case of an exception while waiting for the chain to be at a certain height, we just cut our losses and sync from the current height.
                         this.logger.LogError("Exception occurred while waiting for chain to download: {0}.", ex.Message);
                         this.StartSync(this.chain.Tip.Height);
+
+                        // Set the asyncloop to null so that it could be used again.
+                        this.syncFromDateAsyncLoop = null;
```

Comment:
This is quite hackey because you are not disposing it which means you are relying on specific implementation in which you know what the dispose does. It suggests that the overall design is probably not optimal.